src/components/textbox/__next__/text-input.stories.tsx

```diff
@@ -0,0 +1,345 @@
+    ...styledSystemProps,
+  },
+};
+
+export default meta;
+
+export const Default = () => {
+  const [state, setState] = useState("");
+
+  const setValue = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
+    setState(target.value);
+  };
+
+  return <TextInput label="TextInput" value={state} onChange={setValue} />;
+};
+Default.storyName = "Default";
+
+export const Required = () => {
+  const [state, setState] = useState("");
+
+  const setValue = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
+    setState(target.value);
+  };
+
+  return (
+    <TextInput label="TextInput" required value={state} onChange={setValue} />
+  );
+};
+
+Required.storyName = "Required";
+
+export const Placeholder = () => {
+  const [state, setState] = useState("");
+
+  const setValue = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
+    setState(target.value);
+  };
+
+  return (
+    <TextInput
+      label="TextInput"
+      placeholder="Placeholder text"
+      value={state}
+      onChange={setValue}
+    />
+  );
+};
+
+Placeholder.storyName = "Placeholder";
+Placeholder.parameters = {
+  chromatic: { disableSnapshot: false },
+};
+
+export const WithInputHint = () => {
+  const [state, setState] = useState("");
+
+  const setValue = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
+    setState(target.value);
+  };
+
+  return (
+    <TextInput
+      label="TextInput"
+      inputHint="Hint text (optional)."
+      value={state}
+      onChange={setValue}
+    />
+  );
+};
+
+WithInputHint.storyName = "With Input Hint";
+
+export const WithInputIcon = () => {
+  const [state, setState] = useState("");
+
+  const setValue = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
+    setState(target.value);
+  };
+
+  return (
+    <TextInput
+      inputIcon="placeholder"
+      label="TextInput"
+      value={state}
+      onChange={setValue}
+    />
+  );
+};
+
+WithInputIcon.storyName = "With Input Icon";
+
+export const Disabled = () => {
+  return (
+    <TextInput
+      label="TextInput"
+      disabled
+      placeholder="Placeholder text"
+      value="Disabled value"
+      onChange={() => {}}
+    />
+  );
+};
+
+Disabled.storyName = "Disabled";
+
+export const ReadOnly = () => {
+  return (
+    <TextInput
+      label="TextInput"
+      readOnly
+      placeholder="Placeholder text"
+      value="Read-only value"
+      onChange={() => {}}
+    />
+  );
+};
+
+ReadOnly.storyName = "Read-Only";
+
+export const LabelInline = () => {
+  const [state, setState] = useState("");
+
+  const setValue = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
+    setState(target.value);
+  };
+
+  return (
+    <TextInput
+      label="TextInput"
+      labelInline
+      inputHint="hello world"
+      placeholder="TextInput"
+      value={state}
+      onChange={setValue}
+    />
+  );
+};
+
+LabelInline.storyName = "Label Inline";
+
+export const InputWidth = () => {
+  const [state, setState] = useState("");
```

Comment:
suggestion (non-blocking): might be worth creating state and handlers for each input as I suspect someone may flag that they don't work independently